### PR TITLE
Call the az diagnostic in more context

### PR DESCRIPTION
**Pull request by mpagot**
Move the az command to collect the diagnostic to a dedicated qesap lib function. Call this function also at the end of each test. Add timeout default in an ansible API.
Add docstring for a not documented argument.
More Unit testing.

- Related ticket: [TEAM-8893](https://jira.suse.com/browse/TEAM-8893)

# Verification run:

## qesap

 http://openqaworker15.qa.suse.cz/tests/275014 :green_circle: in case of a EverythingPass test the diagnostic log is collected at the end of the `test_cluster` test module (at least for the qesap regression schedule) --> http://openqaworker15.qa.suse.cz/tests/275014#step/test_cluster/540 
 http://openqaworker15.qa.suse.cz/tests/275016 :green_circle: diagnostic is also collected in case of failure in any other test place http://openqaworker15.qa.suse.cz/tests/275016#step/test_cluster/535
 http://openqaworker15.qa.suse.cz/tests/275017
 http://openqaworker15.qa.suse.cz/tests/275018
 http://openqaworker15.qa.suse.cz/tests/275019
 http://openqaworker15.qa.suse.cz/tests/275020
 http://openqaworker15.qa.suse.cz/tests/275021 : :green_circle:  in case of Ansible failure at RebootTimeout, the code behaves as before: it collects the diagnostic log http://openqaworker15.qa.suse.cz/tests/275021#step/deploy/129

### Azure

http://openqaworker15.qa.suse.cz/tests/275015 :green_circle:  (or :red_circle: ??) in case of HanaSR PASS the diagnostic is not collected.

### CODE REVIEW

File: lib/qesapdeployment.pm [hunk 1]
```diff
@@ -697,6 +698,7 @@ sub qesap_ansible_cmd {
     foreach (qw(provider cmd)) { croak "Missing mandatory $_ argument" unless $args{$_}; }
     $args{user} ||= 'cloudadmin';
     $args{filter} ||= 'all';
+    $args{timeout} //= bmwqemu::scale_timeout(90);
```
Comment (mpagot) [inline lib/qesapdeployment.pm:701]: args was already there but not checked as mandatory with the `croak` and no default value was assigned to it

File: lib/qesapdeployment.pm [hunk 2]
```diff
@@ -1131,6 +1133,13 @@ sub qesap_cluster_logs {
             qesap_upload_crm_report(host => $host, provider => $provider, failok => 1);
         }
     }
+    if ($provider eq 'AZURE') {
```
Comment (mpagot) [inline lib/qesapdeployment.pm:1136]: to have the diagnostic logs collected at any run even if it is  not failing

File: lib/qesapdeployment.pm [hunk 3]
```diff
@@ -2136,6 +2169,11 @@ sub qesap_terraform_clean_up_retry {
     Return 0: we manage the failure properly
     Return 1: something went wrong or we do not know what to do with the failure
 
+=over 1
+
+=item B<ERROR_LOG> - error log filename
```
Comment (mpagot) [inline lib/qesapdeployment.pm:2174]: argument added in  https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/18159/  but not documented

File: lib/qesapdeployment.pm [hunk 4]
```diff
@@ -2194,6 +2226,8 @@ sub qesap_terrafom_ansible_deploy_retry {
         record_info('ANSIBLE RETRY PASS');
     }
     else {
+        qesap_file_find_string(file => $args{error_log}, search_string => 'fatal:');
```
Comment (mpagot) [inline lib/qesapdeployment.pm:2229]: let see if it helps openQA reviewers

**Review by lilyeyes — APPROVED** [review]
Ｇenerally lgtm.

**Review by hadeskun — APPROVED** [review]
LGTM

File: lib/qesapdeployment.pm [hunk 5]
```diff
@@ -1131,6 +1133,13 @@ sub qesap_cluster_logs {
             qesap_upload_crm_report(host => $host, provider => $provider, failok => 1);
         }
     }
+    if ($provider eq 'AZURE') {
+        my @diagnostic_logs = qesap_az_diagnostic_log();
+        foreach (@diagnostic_logs) {
+            push(@log_files, $_);
+            qesap_upload_logs();
```
Comment (BillAnastasiadis) [inline lib/qesapdeployment.pm:1140]: Why not to call `upload_logs` once in the end? Like

```
foreach (@diagnostic_logs) {
            push(@log_files, $_);
            }
qesap_upload_logs();
```

But that's just a matter of preference, nothing important

**Review by BillAnastasiadis — APPROVED** [review]
LGTM

**Pull request merged**